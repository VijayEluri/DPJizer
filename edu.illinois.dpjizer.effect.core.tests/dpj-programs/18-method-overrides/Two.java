/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class A<region R1> {
	A() {
	}

	void m(int j) {

	}
}

class B<region R2> extends A<R2> {
	int[]<<this>>#i a in R2;

	B() {
	}

	void m(int i) {
		a[0] = 0;
	}
}