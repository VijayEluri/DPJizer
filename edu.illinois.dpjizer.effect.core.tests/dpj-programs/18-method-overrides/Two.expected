A.A() invokes java.lang.Object.Object()
A.m(int) pure
B.B() invokes A.A() where { R1 -> R2 }
B.m(int) reads R2 writes this overrides A.m(int) where { R1 -> R2 }
